src/transport_ofi.c

```diff
@@ -98,6 +98,56 @@ shmem_internal_mutex_t          shmem_transport_ofi_lock;
 pthread_mutex_t                 shmem_transport_ofi_progress_lock = PTHREAD_MUTEX_INITIALIZER;
 #endif /* ENABLE_THREADS */
 
+/* Temporarily redefine SHM_INTERNAL integer types to their FI counterparts to
+ * translate the DTYPE_* types (defined by autoconf according to system ABI)
+ * into FI types in the table below */
+#define SHM_INTERNAL_INT8   FI_INT8
+#define SHM_INTERNAL_INT16  FI_INT16
+#define SHM_INTERNAL_INT32  FI_INT32
+#define SHM_INTERNAL_INT64  FI_INT64
+#define SHM_INTERNAL_UINT8  FI_UINT8
+#define SHM_INTERNAL_UINT16 FI_UINT16
+#define SHM_INTERNAL_UINT32 FI_UINT32
+#define SHM_INTERNAL_UINT64 FI_UINT64
```

Comment:
Yeah, I apologize for this ugly hack. The DTYPE_* types are mapped (via preprocessor macros in config.h) to SHM_INTERNAL_[U]INT* types (enums) via config.h. We redefine the SHM_INTERNAL_[U]INT* tokens here to FI_* types to project the DTYPE_* types into the FI_* type space. This temporary redefinition is allowed because, as enums, SHM_INTERNAL_[U]INT* types are not preprocessor macros.